Buildscripts: add a docker buildscript

abdallahmehiz opened the pull request:
Implements a few scripts to build mpv-android inside a docker container.
running `docker-build.sh` would create the docker image, mount the mpv-android project from the host system then starts the build process from `docker-entrypoint.sh`.

abdallahmehiz committed:
Buildscripts: add a docker buildscript

kni7e commented:
@sfan5, any update on this?

sfan5 commented:
While I can see mpv-android shipping with a helper to make setting up docker easier I don't see why it's necessary to essentially duplicate half of the build-related scripts to enable this.

> @sfan5, any update on this?

Update on what? If you want to use these scripts you can do that today. No need to wait for me.

abdallahmehiz committed:
refactor: simplify the docker scripts

made the just forward whatever args to `buildall.sh` or `download.sh`.
with `shell` being a special argument to open a shell inside the container.

abdallahmehiz commented:
> While I can see mpv-android shipping with a helper to make setting up docker easier I don't see why it's necessary to essentially duplicate half of the build-related scripts to enable this.

I've rewritten it so all it does is forward download or build commands to `buildall.sh` or `download.sh`.
The only additional thing is the `shell` argument to launch a shell inside the docker image with everything mounted.

kni7e commented:
> Update on what? If you want to use these scripts you can do that today. No need to wait for me.

Just wanted to know if there is a plan to merge this.

abdallahmehiz committed:
fix: docker builds

sfan5 commented on buildscripts/Dockerfile:
```diff
@@ -0,0 +1,55 @@
+
+# Set up environment variables
+ENV DEBIAN_FRONTEND=noninteractive
+ENV ANDROID_HOME=/opt/android-sdk
```
is the user expected to have the sdk installed system-wide? is this a standard path?

abdallahmehiz replied:
No, the standard path is `~/Android/Sdk`.
That's where the Android SDK defaults to.

abdallahmehiz replied:
It's a container, so /opt made more sense to me than a user home path.
